sys/cmake/SvnVersion.cmake

```diff
@@ -0,0 +1,17 @@
+macro(get_svn_revision OutVar)
+    message("Determining SVN revision")
+    find_program(SVNVERSION_PROGRAM name svnversion)
+    if (SVNVERSION_PROGRAM)
+        execute_process(COMMAND "${SVNVERSION_PROGRAM}" "-c"
```

Comment:
Another problem is that svnversion prints EOL at the end of its output, which somehow gets into the variable. As the result, the last double-quote is put on a separate line in svnversion.h. It has to be removed somehow.

Reply:
Adding OUTPUT_STRIP_TRAILING_WHITESPACE parameter to this command removes the damn EOL for me.

Reply:
Sorry for the svnversion troubles. I didn't really get to test this part due to my git-svn setup. So thanks for trouble-shooting it, I'll get the necessary changes in :)

Reply:
Should be working now, hopefully.